src/Strategy/Maven.hs

```diff
@@ -64,29 +66,47 @@ getDeps ::
   ( Has (Lift IO) sig m
   , Has ReadFS sig m
   , CandidateCommandEffs sig m
+  , Has (Reader MavenScopeFilters) sig m
   ) =>
   MavenProject ->
   m DependencyResults
 getDeps (MavenProject closure) = do
+  includeScopeFilters <- asks includeScope
+  excludeScopeFilters <- asks excludeScope
+  let includeScopeFilterSet = scopes includeScopeFilters
+      excludeScopeFilterSet = scopes excludeScopeFilters
+
   (graph, graphBreadth) <- context "Maven" $ getDepsDynamicAnalysis closure <||> getStaticAnalysis closure
+  let filteredGraph = filterMavenDependencyByScope includeScopeFilterSet excludeScopeFilterSet graph
+      newGraph = gmap mavenDependencyToDependency filteredGraph
```

Comment:
[Prefer] Maybe `finalizedGraph` instead of `newGraph`. I think `new` is slightly confusing.